roles/openshift_node/files/bootstrap.yml

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/ansible-playbook
+---
+- hosts: localhost
+  gather_facts: yes
+  vars:
+    origin_dns:
+      file: /etc/dnsmasq.d/origin-dns.conf
+      lines:
+      - regex: ^listen-address
+        state: absent
+      - regex: "^except-interface"
+        line: except-interface=lo
+        after: EOF
+        state: present
+    node_dns:
+      file: /etc/dnsmasq.d/node-dnsmasq.conf
+      lines:
+      - regex: "^server=/in-addr.arpa/127.0.0.1$"
+        line: server=/in-addr.arpa/127.0.0.1
+      - regex: "^server=/cluster.local/127.0.0.1$"
+        line: server=/cluster.local/127.0.0.1
+
+  tasks:
+  - include_vars: openshift_settings.yaml
+
+  - name: set the data for node_dns
+    lineinfile:
+      create: yes
+      insertafter: EOF
+      path: "{{ node_dns.file }}"
+      regexp: "{{ item.regex }}"
+      line: "{{ item.line | default(omit) }}"
+    with_items: "{{ node_dns.lines }}"
+
+  - name: set the data for origin_dns
+    lineinfile:
+      create: yes
+      state: "{{ item.state | default('present') }}"
+      insertafter: "{{ item.after | default(omit) }}"
+      path: "{{ origin_dns.file }}"
+      regexp: "{{ item.regex }}"
+      line: "{{ item.line | default(omit)}}"
+    with_items: "{{ origin_dns.lines }}"
+
+  - when:
+    - openshift_group_type is defined
+    - openshift_group_type != ''
+    block:
+    - name: determine the openshift_service_type
+      stat:
+        path: /etc/sysconfig/atomic-openshift-node
```

Comment:
Do you not have the service type variables from the top level (deployment_type)?  Or is this called in another context?

Reply:
I do have deployment type and will create `service_type` based on `deployment_type`.

Reply:
I attempted to modify this.  Here's what I found.  The script runs on boot.  The variables referenced in this script are meant to in-scope at start-up.  That means that the `openshift_service_type` is defined when building the AMI but not at start up.

We could move this to a template but we would not want it to be processed during AMI creation except for a single variable.  For this reason doing a `stat` on startup and determining service type is much easier than converting to a template, protecting all of the variables but one, then deploying to the node. 

Hopefully this makes sense.